### PR TITLE
[Refactor] Move CBlockFileInfo and CValidationState out of main

### PR DESCRIPTION
**Pull request by Fuzzbawls**
This is the first part of #1209 that moves the CBlockFileInfo and CValidationState classes (and their supporting constants) out of `main.cpp(.h)`

@barrystyle's original commits have been further split to assist in review-ability ease.

I left the refactor of `nMaxMoneyOut` and `CBlockTemplate` included here as they are rather straight forward, however they each now are in their own respective commits.

### CODE REVIEW

**Comment by random-zebra** [comment]
Ref: https://github.com/PIVX-Project/PIVX/pull/1209#discussion_r360828832

My nit was exclusively about `extern const std::string CURRENCY_UNIT` which is defined here (in `amount.*`) but never actually used. It should be used to replace many instances of the explicit string "PIV" in the code, so I suggest we define it when this is addressed (rather than having unused extern variables in the meantime).

**Comment by Fuzzbawls** [comment]
Removed the unused `const` for now. it can be added back in if/when it is actually used/connected to the GUI

File: src/chainparams.h [hunk 1]
```diff
@@ -92,7 +92,6 @@ class CChainParams
     int FutureBlockTimeDrift(const int nHeight) const;
     bool IsValidBlockTimeStamp(const int64_t nTime, const int nHeight) const;
 
-    CAmount MaxMoneyOut() const { return nMaxMoneyOut; }
```
Comment (random-zebra) [inline src/chainparams.h]: Should remove also the definition of the class variable `nMaxMoneyOut` at line 197.

**Comment by random-zebra** [comment]
Also needs rebase due to conflicts with #1290 just merged.

File: src/chainparams.h (hunk 1 above)
Comment (Fuzzbawls) [reply src/chainparams.h]: good catch, done

**Comment by Fuzzbawls** [comment]
rebased and addressed @random-zebra 's comment

File: src/main.cpp [hunk 2]
```diff
@@ -2632,6 +2632,24 @@ bool CheckInputs(const CTransaction& tx, CValidationState& state, const CCoinsVi
...
 }
 
+/** Abort with a message */
+static bool AbortNode(const std::string& strMessage, const std::string& userMessage="")
+{
+    strMiscWarning = strMessage;
+    LogPrintf("*** %s\n", strMessage);
+    uiInterface.ThreadSafeMessageBox(
+        userMessage.empty() ? _("Error: A fatal internal error occured, see debug.log for details") : userMessage,
+        "", CClientUIInterface::MSG_ERROR);
```
Comment (furszy) [inline src/main.cpp]: The second param here is the dialog title, would be good to set it and not leave it empty. Maybe: "Internal Error".

(Stating it for a subsequent PR, not this one.)

**Comment by random-zebra** [comment]
Merging...

**Pull request merged**